In [1]:
import pandas as pd

In [5]:
df=pd.read_csv("laptop_sales.csv")
print("Data set loded")

Data set loded


In [12]:
#to no find rows and columns
df.shape

(1020, 29)

In [11]:
# to get info such as type of data 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1020 entries, 0 to 1019
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Unnamed: 0              1020 non-null   int64  
 1   Name                    1020 non-null   object 
 2   Brand                   1020 non-null   object 
 3   Price                   1020 non-null   int64  
 4   Rating                  1020 non-null   float64
 5   Processor_brand         1020 non-null   object 
 6   Processor_name          1020 non-null   object 
 7   Processor_variant       996 non-null    object 
 8   Processor_gen           891 non-null    float64
 9   Core_per_processor      1008 non-null   float64
 10  Total_processor         573 non-null    float64
 11  Execution_units         573 non-null    float64
 12  Low_Power_Cores         1020 non-null   float64
 13  Energy_Efficient_Units  1020 non-null   int64  
 14  Threads                 972 non-null    

In [16]:
# display 5 rows
df.head(5)

,Unnamed: 0,Name,Brand,Price,Rating,Processor_brand,Processor_name,Processor_variant,Processor_gen,Core_per_processor,...,Graphics_name,Graphics_brand,Graphics_GB,Graphics_integreted,Display_size_inches,Horizontal_pixel,Vertical_pixel,ppi,Touch_screen,Operating_system
0,0,HP Victus 15-fb0157AX Gaming Laptop (AMD Ryzen...,HP,50399,4.30,AMD,AMD Ryzen 5,5600H,5.0,6.0,...,AMD Radeon RX 6500M,AMD,4.0,False,15.6,1920,1080,141.21,True,Windows 11 OS
1,1,Lenovo V15 G4 ‎82YU00W7IN Laptop (AMD Ryzen 3 ...,Lenovo,26690,4.45,AMD,AMD Ryzen 3,7320U,7.0,4.0,...,AMD Radeon Graphics,AMD,NaN,False,15.6,1920,1080,141.21,False,Windows 11 OS
2,2,HP 15s-fq5007TU Laptop (12th Gen Core i3/ 8GB/...,HP,37012,4.65,Intel,Intel Core i3,1215U,12.0,6.0,...,Intel UHD Graphics,Intel,NaN,False,15.6,1920,1080,141.21,False,Windows 11 OS
3,3,Samsung Galaxy Book2 Pro 13 Laptop (12th Gen C...,Samsung,69990,4.75,Intel,Intel Core i5,1240P,12.0,12.0,...,Intel Iris Xe Graphics,Intel,NaN,False,13.3,1080,1920,165.63,False,Windows 11 OS
4,4,Tecno Megabook T1 Laptop (11th Gen Core i3/ 8G...,Tecno,23990,4.25,Intel,Intel Core i3,1115G4,11.0,2.0,...,Intel UHD Graphics,Intel,NaN,False,15.6,1920,1080,141.21,False,Windows 11 OS


In [18]:
# let's remove the unnamed column
df = df.drop("Unnamed: 0",axis=1)

In [19]:
df.columns

Index(['Name', 'Brand', 'Price', 'Rating', 'Processor_brand', 'Processor_name',
       'Processor_variant', 'Processor_gen', 'Core_per_processor',
       'Total_processor', 'Execution_units', 'Low_Power_Cores',
       'Energy_Efficient_Units', 'Threads', 'RAM_GB', 'RAM_type',
       'Storage_capacity_GB', 'Storage_type', 'Graphics_name',
       'Graphics_brand', 'Graphics_GB', 'Graphics_integreted',
       'Display_size_inches', 'Horizontal_pixel', 'Vertical_pixel', 'ppi',
       'Touch_screen', 'Operating_system'],
      dtype='object')

In [25]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
1015    False
1016    False
1017    False
1018    False
1019    False
Length: 1020, dtype: bool

In [23]:
df.duplicated().sum()

np.int64(0)

In [26]:
df.isnull().sum()

Name                        0
Brand                       0
Price                       0
Rating                      0
Processor_brand             0
Processor_name              0
Processor_variant          24
Processor_gen             129
Core_per_processor         12
Total_processor           447
Execution_units           447
Low_Power_Cores             0
Energy_Efficient_Units      0
Threads                    48
RAM_GB                      0
RAM_type                   22
Storage_capacity_GB         0
Storage_type                0
Graphics_name               2
Graphics_brand              2
Graphics_GB               652
Graphics_integreted         2
Display_size_inches         0
Horizontal_pixel            0
Vertical_pixel              0
ppi                         0
Touch_screen                0
Operating_system            0
dtype: int64

In [62]:
df['Processor_variant'].isnull().sum()

np.int64(24)

In [33]:
df[df['Graphics_GB'].isnull()][['Graphics_name','Graphics_integreted']].head(30)

,Graphics_name,Graphics_integreted
1,AMD Radeon Graphics,False
2,Intel UHD Graphics,False
3,Intel Iris Xe Graphics,False
4,Intel UHD Graphics,False
5,Intel Integrated UHD,True
7,Intel UHD Graphics,False
8,Intel Integrated Iris Xe,True
9,Intel Iris Xe,False
10,Intel Graphics,False
11,AMD Radeon Graphics,False


In [37]:
df[df['Graphics_GB'].notnull()][['Graphics_name','Graphics_integreted']].head(30)

,Graphics_name,Graphics_integreted
0,AMD Radeon RX 6500M,False
6,NVIDIA GeForce RTX 4060,False
12,NVIDIA GeForce RTX 3050,False
15,NVIDIA GeForce RTX 3050 Ti,False
16,NVIDIA GeForce RTX 2050,False
19,NVIDIA GeForce RTX 2050,False
21,NVIDIA GeForce RTX 4050,False
26,NVIDIA GeForce RTX 4050,False
29,NVIDIA GeForce RTX 4050,False
30,NVIDIA GeForce GTX 1650,False


In [38]:
df[
    (df['Graphics_GB'].notnull()) &
    (df['Graphics_integreted'] == True)
][['Graphics_name','Graphics_GB','Graphics_integreted']]

,Graphics_name,Graphics_GB,Graphics_integreted
326,Intel Integrated ARC A530M,4.0,True


In [54]:
df[
    (df['Graphics_GB'].notnull()) &
    (df['Graphics_integreted'] == True)
][['Graphics_name','Graphics_GB','Graphics_integreted']]

,Graphics_name,Graphics_GB,Graphics_integreted
326,Intel Integrated ARC A530M,4.0,True


In [58]:
df['Brand'].nunique()

31

In [59]:
df.nunique()

Name                      1020
Brand                       31
Price                      561
Rating                      17
Processor_brand              7
Processor_name              36
Processor_variant          125
Processor_gen               13
Core_per_processor          12
Total_processor              8
Execution_units              5
Low_Power_Cores              2
Energy_Efficient_Units       2
Threads                     14
RAM_GB                      10
RAM_type                    12
Storage_capacity_GB         10
Storage_type                 4
Graphics_name              136
Graphics_brand               7
Graphics_GB                  6
Graphics_integreted          2
Display_size_inches         22
Horizontal_pixel            22
Vertical_pixel              22
ppi                         62
Touch_screen                 2
Operating_system            13
dtype: int64

In [61]:
df[df['Graphics_name'].isnull()]

,Name,Brand,Price,Rating,Processor_brand,Processor_name,Processor_variant,Processor_gen,Core_per_processor,Total_processor,...,Graphics_name,Graphics_brand,Graphics_GB,Graphics_integreted,Display_size_inches,Horizontal_pixel,Vertical_pixel,ppi,Touch_screen,Operating_system
648,Huawei Qingyun L540 Laptop (Kirin 9006C/ 16GB/...,Huawei,81999,4.3,HiSilicon,HiSilicon Kirin 9006C 9006C,9006C,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,14.0,2160,1440,185.43,False,Linux OS
998,Dell Vostro 5630 Laptop (13th Gen Core i7/ 16G...,Dell,94990,4.6,Intel,Intel Core i7,1355U,13.0,10.0,2.0,...,NaN,NaN,NaN,NaN,16.0,1920,1080,137.68,False,Windows 11 OS


Graphics-related attributes were missing for two records. Since the remaining laptop specifications were available and the missing values affected only a small portion of the dataset (2 out of 1020 rows), the records were retained to avoid unnecessary data loss.

In [63]:
df[df['Total_processor'].isnull()]

,Name,Brand,Price,Rating,Processor_brand,Processor_name,Processor_variant,Processor_gen,Core_per_processor,Total_processor,...,Graphics_name,Graphics_brand,Graphics_GB,Graphics_integreted,Display_size_inches,Horizontal_pixel,Vertical_pixel,ppi,Touch_screen,Operating_system
0,HP Victus 15-fb0157AX Gaming Laptop (AMD Ryzen...,HP,50399,4.30,AMD,AMD Ryzen 5,5600H,5.0,6.0,NaN,...,AMD Radeon RX 6500M,AMD,4.0,False,15.6,1920,1080,141.21,True,Windows 11 OS
1,Lenovo V15 G4 ‎82YU00W7IN Laptop (AMD Ryzen 3 ...,Lenovo,26690,4.45,AMD,AMD Ryzen 3,7320U,7.0,4.0,NaN,...,AMD Radeon Graphics,AMD,NaN,False,15.6,1920,1080,141.21,False,Windows 11 OS
4,Tecno Megabook T1 Laptop (11th Gen Core i3/ 8G...,Tecno,23990,4.25,Intel,Intel Core i3,1115G4,11.0,2.0,NaN,...,Intel UHD Graphics,Intel,NaN,False,15.6,1920,1080,141.21,False,Windows 11 OS
10,Wings Nuvobook V1 Laptop (11th Gen Core i5/ 8G...,Wings,24990,4.70,Intel,Intel Core i5,1135G7,11.0,4.0,NaN,...,Intel Graphics,Intel,NaN,False,15.6,1080,1920,141.21,False,Windows 11 OS
11,Lenovo V15 82KDA01BIH Laptop (AMD Ryzen 3 5300...,Lenovo,26739,4.35,AMD,AMD Ryzen 3,5300U,5.0,4.0,NaN,...,AMD Radeon Graphics,AMD,NaN,False,15.6,1920,1080,141.21,False,Windows 11 OS
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1004,HP Victus 16-d0302TX Gaming Laptop (11th Gen C...,HP,55999,4.25,Intel,Intel Core i5,11400H,11.0,6.0,NaN,...,NVIDIA GeForce GTX 1650,NVIDIA,4.0,False,16.1,1920,1080,136.83,False,Windows 11 OS
1008,HP 14s-dr3003TU 832T7PA Laptop (Pentium Silver...,HP,26800,4.75,Intel,Intel Pentium Silver,N6000,NaN,4.0,NaN,...,Intel UHD Graphics,Intel,NaN,False,14.0,1366,768,111.94,False,Windows 11 OS
1012,Infinix INBook X2 Slim Series XL23 Laptop (11t...,Infinix,34990,4.55,Intel,Intel Core i5,1155G7,11.0,4.0,NaN,...,Intel Integrated Iris Xe,Intel,NaN,True,14.0,1920,1080,157.35,False,Windows 11 OS
1017,HP Victus 15-fb0131AX Gaming Laptop (AMD Ryzen...,HP,58800,4.40,AMD,AMD Ryzen 5,5600H,5.0,6.0,NaN,...,AMD Radeon RX 5600M,AMD,4.0,False,15.6,1920,1080,141.21,False,Windows 11 OS
